qiita_db/software.py

```diff
@@ -86,32 +102,91 @@ def create(cls, software, name, description, cli_cmd, parameters_table):
+                    "{param_name: [paramter_type, default]}. Found: "
+                    "%s for parameter name %s" % (vals, pname))
+
+            ptype, dflt = vals
+            # Check that the type is one of the supported types
+            supported_types = ['string', 'integer', 'float', 'artifact',
+                               'reference']
+            if ptype not in supported_types and not ptype.startswith('choice'):
+                supported_types.append('choice')
+                raise qdb.exceptions.QiitaDBError(
+                    "Unsupported parameters type '%s' for parameter %s. "
+                    "Supported types are: %s"
+                    % (ptype, pname, ', '.join(supported_types)))
+
+            if ptype.startswith('choice') and dflt is not None:
+                choices = loads(ptype.split(':')[1])
+                if dflt not in choices:
+                    raise qdb.exceptions.QiitaDBError(
+                        "The default value '%s' for the parameter %s is not "
+                        "listed in the available choices: %s"
+                        % (dflt, pname, ', '.join(choices)))
+
+            if dflt is not None:
+                sql_param_values.append([pname, ptype, False, dflt])
+            else:
+                sql_param_values.append([pname, ptype, True, None])
+
         with qdb.sql_connection.TRN:
-            if cls.exists(software, cli_cmd):
+            if cls.exists(software, name):
                 raise qdb.exceptions.QiitaDBDuplicateError(
-                    "command", "software: %d, cli_cmd: %s"
-                               % (software.id, cli_cmd))
+                    "command", "software: %d, name: %s"
+                               % (software.id, name))
+            # Add the command to the DB
             sql = """INSERT INTO qiita.software_command
-                            (name, software_id, description, cli_cmd,
-                             parameters_table)
-                     VALUES (%s, %s, %s, %s, %s)
+                            (name, software_id, description)
+                     VALUES (%s, %s, %s)
                      RETURNING command_id"""
-            sql_params = [name, software.id, description, cli_cmd,
-                          parameters_table]
+            sql_params = [name, software.id, description]
             qdb.sql_connection.TRN.add(sql, sql_params)
             c_id = qdb.sql_connection.TRN.execute_fetchlast()
 
+            # Add the parameters to the DB
+            sql = """INSERT INTO qiita.command_parameter
+                        (command_id, parameter_name, parameter_type, required,
+                         default_value)
+                     VALUES (%s, %s, %s, %s, %s)"""
+            sql_params = [[c_id, pname, p_type, reqd, default]
+                          for pname, p_type, reqd, default in sql_param_values]
+            qdb.sql_connection.TRN.add(sql, sql_params, many=True)
```

Comment:
Just to make sure I understand, if the parameters dictionary is empty, then this line is not going to add any parameters associated with the newly created command, but that's ok, right? Or should we disallow this?